service/history/timerQueueProcessor.go

```diff
@@ -169,6 +156,12 @@ func (t *timerQueueProcessorImpl) NotifyNewTimers(timerTasks []persistence.Task)
 	}
 }
 
+// NotifyNewTimerEvents - Notify the processor about the new standby history events arrival.
+// This should be called each time new timer events arrives, otherwise timers maybe fired unexpected.
+func (t *timerQueueProcessorImpl) NotifyNewTimerEvents() {
```

Comment:
I think we should have the same API for notifying new timers whether it is for active processor or standby processor.  Basically we should have a baseTimerQueueProcessor which has common logic and then activeTimerQueueProcessor and passiveTimerQueueProcessor with their own handling of TimerTasks.  But the way you notify both should be exactly the same.

Reply:
this will be done in the next PR